rust/template/observe/lib.rs

```diff
@@ -0,0 +1,44 @@
+pub trait Observable<T, E>
+where T:Send, E: Send
+{
+    // An Observer subscribes to an Observable to listen to data
+    // emitted by the latter. The Observer stops listening when
+    // the Subscription returned is cancelled.
+    fn subscribe(&mut self, observer: Box<dyn Observer<T, E> + Send>) -> Box<dyn Subscription>;
+}
+
+pub trait Observer<T, E>: Send
+where T:Send, E: Send
+{
+    // Action to perform before data starts coming in from the
```

Comment:
Can you please use documentation style comments `///` for documentation like that, so that we can more easily generate a rust doc style documentation. (here and in many other places)